skale_checks/adapters/watchdog.py

```diff
@@ -0,0 +1,125 @@
+#   -*- coding: utf-8 -*-
+#
+#   This file is part of skale-checks
+#
+#   Copyright (C) 2021-Present SKALE Labs
+#
+#   This program is free software: you can redistribute it and/or modify
+#   it under the terms of the GNU Affero General Public License as published by
+#   the Free Software Foundation, either version 3 of the License, or
+#   (at your option) any later version.
+#
+#   This program is distributed in the hope that it will be useful,
+#   but WITHOUT ANY WARRANTY; without even the implied warranty of
+#   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#   GNU Affero General Public License for more details.
+#
+#   You should have received a copy of the GNU Affero General Public License
+#   along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import re
+from skale_checks.adapters.connectors import (Connector, construct_ok_response,
+                                              construct_err_response, Response)
+
+WATCHDOG_TIMEOUT_DEFAULT = 10
+WATCHDOG_PORT = 3009
+
+
+class WatchdogBase(Connector):
+    def __init__(self, node_ip, timeout=WATCHDOG_TIMEOUT_DEFAULT):
+        self.watchdog_url = get_watchdog_url(node_ip)
+        self.timeout = timeout
+        super().__init__(self.watchdog_url, self.timeout)
+
+    def core_status(self):
```

Comment:
this class could be replaced with a single URL handler function (see skale.py handler for calls)